src/distilabel/distiset.py

```diff
@@ -81,7 +83,26 @@ def push_to_hub(
                 Whether to generate a dataset card or not. Defaults to True.
             **kwargs:
                 Additional keyword arguments to pass to the `push_to_hub` method of the `datasets.Dataset` object.
+
+        Raises:
+            ValueError: If no token is provided and couldn't be retrieved automatically.
         """
+        if token is None:
+            from huggingface_hub import constants
+
+            if token := os.getenv(_INFERENCE_ENDPOINTS_API_KEY_ENV_VAR_NAME):
+                pass
+            if token is None:
+                if not Path(constants.HF_TOKEN_PATH).exists():
+                    raise ValueError(
+                        f"To use `{self.__class__.__name__}` an API key must be provided via"
+                        " `token`, set the environment variable"
+                        f" `{_INFERENCE_ENDPOINTS_API_KEY_ENV_VAR_NAME}` or use the `huggingface-hub` CLI to login"
+                        " with `huggingface-cli login`."
+                    )
+                with open(constants.HF_TOKEN_PATH) as f:
+                    token = f.read().strip()
```

Comment:
could we move this logic to `utils.huggingface` or similar? we could reuse this function in `inference_endpoints.py`